StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/MaintainabilityRules/SA1653CodeFixProvider.cs

```diff
@@ -0,0 +1,129 @@
+                case SyntaxKind.CollectionInitializerExpression:
+                    newParent = RewriteInitializer((InitializerExpressionSyntax)parent);
+                    break;
+
+                case SyntaxKind.AnonymousObjectCreationExpression:
+                    newParent = RewriteAnonymousObjectInitializer((AnonymousObjectCreationExpressionSyntax)parent);
+                    break;
+
+                default:
+                    throw new InvalidOperationException("Unknown initializer type: " + parent.Kind());
+            }
+
+            var newSyntaxRoot = syntaxRoot.ReplaceNode(parent, newParent);
+
+            var newDocument = document.WithSyntaxRoot(newSyntaxRoot.WithoutFormatting());
+            return newDocument;
+        }
+
+        private static SyntaxNode RewriteInitializer(InitializerExpressionSyntax initializer)
+        {
+            var existingItems = new List<ExpressionSyntax>(initializer.Expressions);
+            var last = existingItems.Last();
+            existingItems.Remove(last);
+            existingItems.Add(last.WithTrailingTrivia(last.GetTrailingTrivia().WithoutTrailingWhitespace()));
+
+            var existingSeparators = initializer.Expressions.GetSeparators();
+            var newSeparators = new List<SyntaxToken>(existingSeparators);
+            newSeparators.Add(SyntaxFactory.Token(SyntaxKind.CommaToken).WithTrailingTrivia(last.GetTrailingTrivia()));
+
+            var newInitializerExpressions = SyntaxFactory.SeparatedList(
+                existingItems,
+                newSeparators);
+
+            var newInitializer = SyntaxFactory.InitializerExpression(
+                initializer.Kind(),
+                initializer.OpenBraceToken,
+                newInitializerExpressions,
+                initializer.CloseBraceToken);
+
+            return newInitializer;
+        }
+
+        private static SyntaxNode RewriteAnonymousObjectInitializer(AnonymousObjectCreationExpressionSyntax initializer)
+        {
+            var existingItems = new List<AnonymousObjectMemberDeclaratorSyntax>(initializer.Initializers);
+            var last = existingItems.Last();
+            existingItems.Remove(last);
+            existingItems.Add(last.WithTrailingTrivia(last.GetTrailingTrivia().WithoutTrailingWhitespace()));
+
+            var existingSeparators = initializer.Initializers.GetSeparators();
+            var newSeparators = new List<SyntaxToken>(existingSeparators);
+            newSeparators.Add(SyntaxFactory.Token(SyntaxKind.CommaToken).WithTrailingTrivia(last.GetTrailingTrivia()));
+
+            var newInitializerExpressions = SyntaxFactory.SeparatedList(
+                existingItems,
+                newSeparators);
+
+            var newInitializer = SyntaxFactory.AnonymousObjectCreationExpression(
+                initializer.NewKeyword,
+                initializer.OpenBraceToken,
+                newInitializerExpressions,
+                initializer.CloseBraceToken);
```

Comment:
:thought_balloon: This should use `initializer.WithInitializers(newInitializerExpressions)`.